chainer/dataset/convert.py

```diff
@@ -0,0 +1,114 @@
+import numpy
+import six
+
+from chainer import cuda
+
+
+def concat_examples(batch, device=None, padding=None):
+    """Concatenates a list of examples into array(s).
+
+    Dataset iterator yields a list of examples. If each example is an array,
+    this function concatenates them along the newly-inserted first axis (called
+    `batch dimension`) into one array. The basic behavior is same for examples
+    consisting of multiple arrays, i.e., corresponding arrays of all examples
+    are concatenated.
+
+    For instance, consider each example consists of two arrays ``(x, y)``.
```

Comment:
I read the first paragraph again and come to think it can specify the behavior of this function. So you do not have to change it.